File: python/structures/test/sorts/bubble_sort_test.py
import structures.src.sorts.bubble as bubble_sort
from structures.src.util.sort_util import random_array, is_sorted
import structures.src.util.constants as const


def test_sorted():
    arr = random_array(const.ARRAY_SIZE)

    # make sure the starting array isn't sorted
    assert not is_sorted(arr)

    # sort the array
    result = bubble_sort.sort(arr)

    # make sure it worked
    assert is_sorted(result)

File: go/src/sorts/bubblesort_test.go
package sorts

import "../testhelper"

import "testing"

func TestBubbleSort(t *testing.T) {
  arr := testhelper.RandomIntSlice(testhelper.ArraySize)

  sorted := BubbleSort(arr)

  testhelper.AssertIsSorted(sorted, t)
}
